pkg/controller/seed-controller-manager/constraint-controller/controller.go

```diff
@@ -0,0 +1,280 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package seedconstraintsynchronizer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+
+	kubermaticapiv1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	controllerutil "k8c.io/kubermatic/v2/pkg/controller/util"
+	"k8c.io/kubermatic/v2/pkg/controller/util/predicate"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	kuberneteshelper "k8c.io/kubermatic/v2/pkg/kubernetes"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+	"k8c.io/kubermatic/v2/pkg/util/workerlabel"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	// This controller syncs the kubermatic constraints to constraint on the user cluster.
+	ControllerName = "constraint_syncing_controller"
+	finalizer      = kubermaticapiv1.GatekeeperConstraintCleanupFinalizer
+)
+
+type reconciler struct {
+	log                     *zap.SugaredLogger
+	workerNameLabelSelector labels.Selector
+	recorder                record.EventRecorder
+	namespace               string
+	seedClient              ctrlruntimeclient.Client
+}
+
+func Add(ctx context.Context,
+	mgr manager.Manager,
+	log *zap.SugaredLogger,
+	workerName string,
+	namespace string,
+	numWorkers int) error {
+
+	workerSelector, err := workerlabel.LabelSelector(workerName)
+	if err != nil {
+		return fmt.Errorf("failed to build worker-name selector: %v", err)
+	}
+
+	reconciler := &reconciler{
+		log:                     log.Named(ControllerName),
+		workerNameLabelSelector: workerSelector,
+		recorder:                mgr.GetEventRecorderFor(ControllerName),
+		namespace:               namespace,
+		seedClient:              mgr.GetClient(),
+	}
+
+	c, err := controller.New(ControllerName, mgr, controller.Options{Reconciler: reconciler, MaxConcurrentReconciles: numWorkers})
+	if err != nil {
+		return fmt.Errorf("failed to construct controller: %v", err)
+	}
+
+	if err := c.Watch(
+		&source.Kind{Type: &kubermaticv1.Cluster{}},
+		enqueueConstraints(reconciler.seedClient, reconciler.log, namespace),
+		workerlabel.Predicates(workerName),
+	); err != nil {
+		return fmt.Errorf("failed to create watch for clusters: %w", err)
+	}
+
+	if err := c.Watch(
+		&source.Kind{Type: &kubermaticv1.Constraint{}},
+		&handler.EnqueueRequestForObject{},
+		predicate.ByNamespace(namespace),
```

Comment:
We should also reconcile when the `Constraints` change on the cluster namespaces, isn't it?

Reply:
We discussed this with @imharshita , and for this case it seems to us that the default 10h resync will be ok. 

The reason is that users anyway wont be able to edit/delete these "default/global" constraints (at least from the API side, if they have kubeconfig access then its a different story), so there is no need to add active protection from that. Let us know if you see something we are missing

Reply:
Well, of course in the happy path everything works fine :-). This would allow to automatically recover when `Constraints` get accidentally edited or removed. I don't think should be too complicated to put in place. Isn't it?

Reply:
no, acutally we could add a separate watch for Constraints on the user-cluster namespaces which have some default label, so we dont watch the "user" constraints changes. I just dont know anymore when to rely on the resync and when to watch actively in KKP :D. 

BTW @imharshita , this reminded me that it would be good to add the labeling of the "default" constraints in this PR, so we can distinguish them later in the API.